internal/login/client_test.go

```diff
@@ -0,0 +1,57 @@
+package login_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/launchdarkly/ldcli/internal/login"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMakeRequest(t *testing.T) {
+	t.Run("with a successful response", func(t *testing.T) {
+		server := makeServer(t, http.StatusOK, `{"message": "success"}`)
+		defer server.Close()
+		c := login.NewClient("test-version")
+
+		response, err := c.MakeRequest("POST", server.URL, []byte(`{}`))
+
+		require.NoError(t, err)
+		assert.JSONEq(t, `{"message": "success"}`, string(response))
+	})
+
+	t.Run("with an error with a response body", func(t *testing.T) {
+		server := makeServer(t, http.StatusBadRequest, `{"message": "an error"}`)
+		defer server.Close()
+		c := login.NewClient("test-version")
+
+		_, err := c.MakeRequest("POST", server.URL, []byte(`{}`))
+
+		assert.EqualError(t, err, `{"message": "an error"}`)
+	})
+
+	t.Run("with a method not allowed status code and no body", func(t *testing.T) {
```

Comment:
The other two tests are backfill, and this is a new one. I went with httptest since we don't own the `http.Client` and wanted to keep the test code as real as possible. Since we own the `UnauthenticatedClient` interface, we mock that elsewhere.